doom/config.org
* редакторНормальная прокрутка#+begin_src emacs-lisp;; (setq scroll-margin 1;;   scroll-step 1;;   ;; scroll-conservatively 100000;;   scroll-preserve-screen-position 1;;   mouse-wheel-progressive-speed nil;;   mouse-wheel-scroll-amount '(1 ((shift) . 1)));; (require 'fast-scroll);; (add-hook 'fast-scroll-start-hook (lambda () (flycheck-mode -1)));; (add-hook 'fast-scroll-end-hook (lambda () (flycheck-mode 1)));; (fast-scroll-config);; (fast-scroll-mode 1)(if (>= emacs-major-version 29)  (pixel-scroll-precision-mode 1)    (setq pixel-scroll-precision-large-scroll-height 40.0))#+end_srcДелаем title bar по-уже#+begin_src emacs-lisp(tool-bar-mode 1)(tool-bar-mode 0)#+end_srcСкрываем номера строк#+begin_src emacs-lisp(setq display-line-numbers-type nil)#+end_srcДелаем курсор нормальным (не квадратным, как в терминале)#+begin_src emacs-lisp(setq-default cursor-type '(bar . 2))#+end_srcВключаем горизонтальный скролл#+begin_src emacs-lisp(setq mouse-wheel-tilt-scroll t      mouse-wheel-flip-direction t)#+end_srcЦентрируем код#+begin_src emacs-lisp(setq cwm-centered-window-width 110)(centered-window-mode t)(after! doom-modeline  (setq mode-line-right-align-edge 'right-fringe))#+end_srcНастройки шрифта#+begin_src emacs-lisp(setq doom-font (font-spec :family "FiraCode Nerd Font Mono" :size 14 :weight 'normal))(add-hook! 'prog-mode-hook  (setq-local default-text-properties '(line-spacing 0.2 line-height 1.3)))#+end_srcИзменяем формат подсветки ошибок#+begin_src emacs-lisp(custom-set-faces!  '(flycheck-error :background "#FFE0E0" :underline nil))#+end_srcОтключаем жирный шрифт#+begin_src emacs-lisp(after! doom-themes  (setq doom-themes-enable-bold nil))#+end_srcКрасивые символы#+begin_src emacs-lisp(plist-put! +ligatures-extra-symbols  :name          "»"  :src_block     "»"  :src_block_end "«"  :quote         "“"  :quote_end     "”"  :lambda        "λ"  :def           "ƒ"  :composition   "∘"  :map           "↦"  :null          "∅"  :true          nil  :false         nil  :int           "ℤ"  :float         "ℝ"  :str           nil  :bool          "𝔹"  :list          nil  :not           "￢"  :in            "∈"  :not-in        "∉"  :and           "∧"  :or            "∨"  :for           "∀"  :some          "∃"  :return        ""  :yield         ""  :union         "⋃"  :intersect     "∩"  :diff          "∖"  :tuple         "⨂"  :pipe          ""  :dot           "•")#+end_srcНастройка темы#+begin_src emacs-lisp(setq doom-theme 'doom-alabaster)#+end_srcM-x во высплывающем окне#+begin_src emacs-lisp(require 'vertico-posframe)(vertico-posframe-mode 1)(setq vertico-posframe-parameters      '((left-fringe . 16)        (right-fringe . 16)))#+end_srcАвтосохранение#+begin_src emacs-lisp(setq auto-save-default t      make-backup-files t)#+end_src* ХоткеиУдалить текущую строку#+begin_src emacs-lisp(map! "s-<backspace>" #'kill-whole-line)#+end_srcЗакрытие вкладок по хоткею command + w#+begin_src emacs-lisp(map! "s-w" #'kill-this-buffer)#+end_srcОткрытие дерева по хоткею command + 1#+begin_src emacs-lisp(map! "s-1" #'treemacs)#+end_srcОткрытие терминала по хоткею command + 3#+begin_src emacs-lisp(map! "s-3" #'+vterm/toggle)#+end_srcПоиск везде по command + shift + f#+begin_src emacs-lisp(map! "s-F" #'+vertico/project-search)#+end_srcДублирование строки по command + d#+begin_src emacs-lisp(map! "s-d" #'duplicate-line)#+end_srcОткрытие докера по command + 8#+begin_src emacs-lisp(map! "s-8" #'docker-containers)#+end_srcСворачивание кода#+begin_src emacs-lisp(global-origami-mode)(map! "s-." #'origami-toggle-node)#+end_srcПерейти к определению#+begin_src emacs-lisp(map! "s-<mouse-1>" #'lsp-goto-type-definition)#+end_srcПерейти в предыдущее положение курсора#+begin_src emacs-lisp(map! "M-[" #'pop-global-mark)#+end_src* orgРазные шрифты для разных заголовков#+begin_src emacs-lisp(custom-set-faces '(org-document-title ((t (:inherit outline-1 :height 2.0))))  '(org-level-1 ((t (:inherit outline-1 :height 1.75))))  '(org-level-2 ((t (:inherit outline-2 :height 1.4))))  '(org-level-3 ((t (:inherit outline-3 :height 1.25))))  '(org-level-4 ((t (:inherit outline-4 :height 1.1)))))#+end_src* Дерево файловодинарный клик для открытия файлов#+begin_src emacs-lisp(with-eval-after-load 'treemacs  (define-key treemacs-mode-map [mouse-1] #'treemacs-single-click-expand-action))#+end_srcИнтеграция дерева с projectile#+begin_src emacs-lisp(after! (treemacs projectile)  (treemacs-project-follow-mode 1))#+end_src* LSPПодсчет количества использований#+begin_src emacs-lisp(setq lsp-lens-enable t)#+end_srcДокументация при наведении мыши#+begin_src emacs-lisp(setq lsp-ui-doc-show-with-cursor t)#+end_src* Программирование** python#+begin_src emacs-lisp#+end_src* Константы#+begin_src emacs-lisp(setq user-full-name "w0rng"      user-mail-address "me@w0rng.ru")#+end_src

configs/doom/config.el
;; (setq scroll-margin 1
;;   scroll-step 1
;;   ;; scroll-conservatively 100000
;;   scroll-preserve-screen-position 1
;;   mouse-wheel-progressive-speed nil
;;   mouse-wheel-scroll-amount '(1 ((shift) . 1)))

;; (require 'fast-scroll)
;; (add-hook 'fast-scroll-start-hook (lambda () (flycheck-mode -1)))
;; (add-hook 'fast-scroll-end-hook (lambda () (flycheck-mode 1)))
;; (fast-scroll-config)
;; (fast-scroll-mode 1)

(if (>= emacs-major-version 29)
  (pixel-scroll-precision-mode 1)
    (setq pixel-scroll-precision-large-scroll-height 40.0))

(tool-bar-mode 1)
(tool-bar-mode 0)

(setq display-line-numbers-type nil)

(setq-default cursor-type '(bar . 2))

(setq mouse-wheel-tilt-scroll t
      mouse-wheel-flip-direction t)

(setq cwm-centered-window-width 110)
(centered-window-mode t)

(after! doom-modeline
  (setq mode-line-right-align-edge 'right-fringe))

(setq doom-font (font-spec :family "FiraCode Nerd Font Mono" :size 14 :weight 'normal))
(add-hook! 'prog-mode-hook
  (setq-local default-text-properties '(line-spacing 0.2 line-height 1.3)))

(custom-set-faces!
  '(flycheck-error :background "#FFE0E0" :underline nil))

(after! doom-themes
  (setq doom-themes-enable-bold nil))

(plist-put! +ligatures-extra-symbols
  :name          "»"
  :src_block     "»"
  :src_block_end "«"
  :quote         "“"
  :quote_end     "”"
  :lambda        "λ"
  :def           "ƒ"
  :composition   "∘"
  :map           "↦"
  :null          "∅"
  :true          nil
  :false         nil
  :int           "ℤ"
  :float         "ℝ"
  :str           nil
  :bool          "𝔹"
  :list          nil
  :not           "￢"
  :in            "∈"
  :not-in        "∉"
  :and           "∧"
  :or            "∨"
  :for           "∀"
  :some          "∃"
  :return        ""
  :yield         ""
  :union         "⋃"
  :intersect     "∩"
  :diff          "∖"
  :tuple         "⨂"
  :pipe          ""
  :dot           "•")

(setq doom-theme 'doom-alabaster)

(require 'vertico-posframe)
(vertico-posframe-mode 1)
(setq vertico-posframe-parameters
      '((left-fringe . 16)
        (right-fringe . 16)))

(setq auto-save-default t
      make-backup-files t)

(map! "s-<backspace>" #'kill-whole-line)

(map! "s-w" #'kill-this-buffer)

(map! "s-1" #'treemacs)

(map! "s-3" #'+vterm/toggle)

(map! "s-F" #'+vertico/project-search)

(map! "s-d" #'duplicate-line)

(map! "s-8" #'docker-containers)

(global-origami-mode)
(map! "s-." #'origami-toggle-node)

(map! "s-<mouse-1>" #'lsp-goto-type-definition)

(map! "M-[" #'pop-global-mark)

(custom-set-faces
 '(org-document-title ((t (:inherit outline-1 :height 2.0))))
  '(org-level-1 ((t (:inherit outline-1 :height 1.75))))
  '(org-level-2 ((t (:inherit outline-2 :height 1.4))))
  '(org-level-3 ((t (:inherit outline-3 :height 1.25))))
  '(org-level-4 ((t (:inherit outline-4 :height 1.1)))))

(with-eval-after-load 'treemacs
  (define-key treemacs-mode-map [mouse-1] #'treemacs-single-click-expand-action))

(after! (treemacs projectile)
  (treemacs-project-follow-mode 1))

(setq lsp-lens-enable t)

(setq lsp-ui-doc-show-with-cursor t)



(setq user-full-name "w0rng"
      user-mail-address "me@w0rng.ru")
